### ISSUE TITLE
Cullenect Labels V2 Implementation

### ISSUE THREAD

**Issue by CullenJWebb** [ndevenish/gflabel#7]
Hello,

In discussion with [Gridfinity Extended OpenSCAD the idea arose to change the orientation of the click interface](https://github.com/ostat/gridfinity_extended_openscad/issues/20), the 3 ribs, from vertical to horizontal. This would allow for labels of any width similar to the Pred labels.

However, besides this being a breaking change for anyone who has already printed their bins, this Python library would no longer be usable for generating those labels. I have come to depend on this library of yours to such a degree that I would not be interested in making a change to the click label standard without your tool also being updated.

**So before I continue with this I have a few questions:**

1. Would you be up to the task of updating this tool if I do in fact create a v2 of the click label standard?
2. If so, are there any requests or guidance that I should keep in mind during the redesign that would make this task easier for you?
3. Also, would it be possible to maintain support for the old labels in some capacity for everyone who has already printed their bins? No future updates/features would need to be supported, just the basic text + icon functionality already present.
4. Could the upcoming v2 "Webb Labels" be renamed to "Cullenect" labels? Bit of a vanity request there, but I plan on rebranding them to be more easily distinguished from the many "click" baseplates and lids already in the Gridfinity ecosystem.

**Here is what I have in mind for the v2 "Cullenect" label interface:**

1. Keep the same profile for the ribs and the slots, only rotated 90⁰ to be horizontal.
2. Reduce rib count from 3 to 2
3. The default 1u width label may be reduced in size from 36.4mm down to a flat 36mm or even 33mm. In testing I would also decide upon the recommended formula for label widths of any given number of units, even those less wide than 1u.
4. I may add a way to easily remove the labels with a corner recess in the slot.
5. The Z thickness of the labels may be increased from 1mm to 1.2mm to reduce the chance of the ribs being visible on the front of the label when using some colors. This is not at all guaranteed though as keeping the labels as thin as possible is important to me.

**Comment by CullenJWebb** [comment]
I've attached here the work in progress files in case that helps to visualize the changes I plan to make.

One unforeseen benefit to the horizontal ribs is that the vertical socket is much easier to print. I haven't packaged that in this preview files though.

[V2.0.3 BETA.zip](https://github.com/user-attachments/files/16134814/V2.0.3.BETA.zip)

**Comment by ostat** [comment]
I think this looks good. A couple of comments.
1. The new design requires supports, where the old design, had a smaller overhang.
2. Instead of having the clips down the middle could the go down the edges? We could then prob angle up the roof to remove the the overhang.

Something like this as an example. The see through boxes are to be removed, hopefully you get what I mean.
![image](https://github.com/ndevenish/gflabel/assets/2128234/78b73029-983e-4656-b3d2-6cfb7719cf98)

**Comment by CullenJWebb** [comment]
@ostat I will give that a shot and report back my findings.

One reason I didn't go for the edge clips is for the vertical slots. With the slots being on the edge I wouldn't be able to print the rib in the vertical slot for the top latch. With the latches where they are now I can print a 45 degree support up into the latch.

However, I am confident that the old latches and the new latches are identical in size, I used the same sketch profile and only changed the sketch plane to the narrower side of the labels. I did tighten the tolerance of the ribs in the slot a bit, but that wouldn't affect the label.

I'll double check that the measurements didn't change while I'm in there testing the edge latches.

**Comment by ndevenish** [comment]
Oh! I am sorry, for some reason github was not sending me notifications for this repository, and I've been travelling for work the past few weeks. I'm of course happy to change things/add options. I'll try to look at this over the next week (off again for a conference next week...)

**Comment by ndevenish** [comment]
And back... (allowing for inevitable COVID illness after conference).

I've started work on these in https://github.com/ndevenish/gflabel/pull/8 . 

I see that https://makerworld.com/en/models/446624 is still on the old v1, so you haven't released yet; Has your design evolved from this beta-version yet?

**Comment by ndevenish** [comment]
In terms of answering these questions:
- Yes, supporting multiple versions is fine (It seems unlikely the old versions will change a lot...)
- I don't particularly need anything except for the fact that I don't currently retain a lot of build123d memory in my head, so would prefer to wait for something relatively close to final before trying to reproduce the geometry (that said, the complicated parts about v1.1 was the geometry on the rib holes, and it looks from above screenshot to be the same, so maybe not so much work..)

**Comment by CullenJWebb** [comment]
Sorry about the delay in response from me. I'm buying a house and things have been upheaved, but this is still a priority for me as a lot of people are using these. My next tests:

- Edge mounted labels as suggested by ostat. These would need to be a massive improvement to be worth it as a vertical slot would no longer be printable without supports.
- 1U wide labels with both vertical and horizontal sockets to fit in either V1 or V2 bins. I think this would be printable but I need to make sure. If this works then no special effort would need to be made for everyone who already printed bins with V1 sockets in them.

**Comment by ndevenish** [comment]
No worries, no pressure. I took several weeks to respond (and I know how stressful house buying can be).

**Comment by CullenJWebb** [comment]
@ndevenish The 2.0 version of the labels are ready here: https://github.com/CullenJWebb/Cullenect-Labels

They are being renamed "Cullenect" to differentiate between other swappable label types. Here are some of the features:

- Backward compatible with v1 bins (when generating 1u width labels)
- Variable label width
- Simpler geometry, easier to print and model.
- Emboss or deboss text
- More satisfying click
- Labels smaller than the socket can slide left and right along the new rail socket design
- Better vertical slot

And best of all it's been designed in OpenSCAD so that basic labels can be generated by users. Icons are still difficult in OpenSCAD though so I'm going to help get this implemented in gflabel as well.

This is my first time modeling in OpenSCAD so I'm sure there are a lot of improvements that could be made to the script itself, but the geometery is set.

**Comment by CullenJWebb** [comment]
@ndevenish Just a heads up that some drawings were added to [the github page](https://github.com/CullenJWebb/Cullenect-Labels) to help outline the new geometry. The drawings are not done yet for the backwards compatible version, but they are similar and can be generated in OpenSCAD.

**Comment by ndevenish** [comment]
> @ndevenish Just a heads up that some drawings were added to [the github page](https://github.com/CullenJWebb/Cullenect-Labels) to help outline the new geometry. The drawings are not done yet for the backwards compatible version, but they are similar and can be generated in OpenSCAD.

Thanks, sorry I didn't get to this yet, this is the first time I've had to spend on "Hobby" programming for a while (babies are time consuming).

I think this version, I like the compatibility and also that the geometry is a bit simpler.

This document and these drawings specify 36 mm wide (42 mm * U - 6 mm) - but the `blank label.step` and F360 sources have it at 36.4 mm - the same as the v1 version. Which is it supposed to be? I also assume you don't support printing "wider" labels that have the backwards-compatibility ribs? (and thus for wider labels they should not be generated...)

**Comment by ndevenish** [comment]
Okay, I think #14 matches the v2 geometry now, only thing to resolve is the width ambiguity (variable-width labels will be easy to add in once this width issue is resolved).

**Comment by CullenJWebb** [comment]
No worries! Being a parent is definitely the priority and I don't expect you to prioritize any of this over your own hobbies. No expectations whatsoever.

`Which is it supposed to be?`

36mm is the new official standard for 1U. I borked the versions I generated with Fusion 360. The OpenSCAD model is the official standard, full stop. I'm re-generating all of the models and will re-upload to Makerworld ASAP.

`I also assume you don't support printing "wider" labels that have the backwards-compatibility ribs?`

Correct. In the OpenSCAD generator you'll see that any width other that 1U (more OR less) results in the backward compatibility being disabled. It can also be disabled manually for more surface area on the first print for people who don't have v1 bins.
